src/ocl/convolutionocl.cpp

```diff
@@ -1591,6 +1540,158 @@ static inline bool IsAlgorithmDisabled(const miopenConvAlgorithm_t algo)
+                                                 const ProblemDescription& problem,
+                                                 const size_t maxSolutionCount,
+                                                 size_t* const solutionCount,
+                                                 miopenConvSolution_t* const solutions) const
+{
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMMED_FALLBACK{}))
+    {
+        MIOPEN_LOG_I("Disabled via environment");
+        *solutionCount = 0;
+        return;
+    }
+
+    /// \todo This is terrible. Should do away when we converge to
+    /// single conv::ProblemDescription type.
+    const auto& inDesc = problem.direction.IsForward() ? problem.conv_problem.GetIn()
+                                                       : problem.conv_problem.GetOut();
+    const auto& outDesc = problem.direction.IsForward() ? problem.conv_problem.GetOut()
+                                                        : problem.conv_problem.GetIn();
+    const auto& weightsDesc = problem.conv_problem.GetWeights();
+    // This check is needed on fallback path only.
+    // On regular path (find-db hit) this was checked during Find().
+    ValidateGroupCount(inDesc, weightsDesc, *this);
+
+    auto i = std::size_t{0};
+
+    std::vector<SolutionSortWrapper> interim;
+    interim.reserve(maxSolutionCount); // For speed. In most cases we have less entries than asked.
+
+    auto ctx = ConvolutionContext{problem};
+    ctx.SetStream(&handle);
+    ctx.DetectRocm();
+
+    const auto& map = miopen::solver::GetMapValueToAnySolver();
+    for(const auto& item : map)
+    {
+        const auto solver_id = solver::Id{item.first};
+        // solver_id is always valid here, because taken from registry.
+        // Validity check is not required.
+        const auto algo = solver_id.GetAlgo();
+        if(IsAlgorithmDisabled(algo)) // Algos can be disabled globally.
+            continue;
+        const auto& s = item.second;
+        if(!s.IsApplicable(ctx))
+            continue;
+        if(!s.IsDynamic()) // Let's allow non-dynamic later, if necessary.
+        {
+            MIOPEN_LOG_I2(solver_id.ToString() << " Not dynamic, skipped");
+            continue;
+        }
+
+        // gemm can appear here only after actual (non-dummy) GEMM Solver is implemented.
+        if(solver_id == solver::Id::gemm())
+            MIOPEN_LOG_W("GEMM solver is ready, rework this function");
+
+        const auto wti = s.GetWti(ctx);
+        MIOPEN_LOG_I2(solver_id.ToString() << " Estimated WTI = " << wti);
+        if(wti <= 0.0f) // Skip unknown WTIs and avoid DIV/0.
+            continue;
+        const auto time = 10.0f / wti; // Assume WTI == 1.0 (100%) is 10 ms.
+
+        interim.emplace_back(time, s.GetWorkspaceSize(ctx), solver_id.Value(), algo);
+    }
+
+    if(!interim.empty())
+    {
+        std::sort(begin(interim), end(interim));
+        for(const auto& entry : interim)
```

Comment:
You can use `std::distance` and some fancy iterator-based copy function.

Reply:
Let me be as simple as possible unless substantial reasons for fancy stuff.